TST: how should we treat internal numpy ufuncs in completion tests ?

context:
https://github.com/numpy/numpy/pull/30984 added a couple "internal" ufuncs to numpy (`np.real` and `np.imag`), which I caught by running astropy's test suite against the dev branch, which failed the completion test for Quantity ufunc helpers

(relevant excerpt from the test logs)
```
>       assert all_np_ufuncs - all_q_ufuncs == set()
E       AssertionError: assert {<ufunc 'imag...ufunc 'real'>} == set()
E         
E         Extra items in the left set:
E         <ufunc 'imag'>
E         <ufunc 'real'>
E         
E         Full diff:
E         - set()
E         + {
E         +     <ufunc 'imag'>,
E         +     <ufunc 'real'>,
E         + }
```

I think the notion of "internal ufuncs" is novel (it's new to me, at least), so I don't know how we want to approach them;
should they still get their own helpers, or should we just special case them in completion tests ?
Since @mhvk was involved in reviewing the patch, he's probably in a great position to answer this question.

--- a/astropy/units/quantity_helper/helpers.py
+++ b/astropy/units/quantity_helper/helpers.py
@@ -18,6 +18,7 @@ from astropy.utils.compat.numpycompat import (
     NUMPY_LT_2_1,
     NUMPY_LT_2_2,
     NUMPY_LT_2_3,
+    NUMPY_LT_2_5,
 )
 
 from . import UFUNC_HELPERS, UNSUPPORTED_UFUNCS
@@ -454,6 +455,9 @@ invariant_ufuncs = (
     np.trunc,
     np.positive,
 )
+if not NUMPY_LT_2_5:
+    invariant_ufuncs += (np_umath.imag, np_umath.real)
+
 for ufunc in invariant_ufuncs:
     UFUNC_HELPERS[ufunc] = helper_invariant
 

